troposphere/__init__.py

```diff
@@ -162,10 +163,51 @@ def validate(self):
         pass
 
     @classmethod
-    def from_dict(cls, title, dict):
-        obj = cls(title)
-        obj.properties.update(dict)
-        return obj
+    def _from_dict(cls, title=None, **kwargs):
+        props = {}
+        for prop_name, value in kwargs.items():
+            try:
+                prop_attrs = cls.props[prop_name]
+            except KeyError:
+                raise AttributeError("Object type %s does not have a "
+                                     "%s property." % (cls.__name__,
+                                                       prop_name))
+            prop_type = prop_attrs[0]
+            if prop_name in kwargs:
+                value = kwargs[prop_name]
+                is_aws_object = False
+                try:
+                    is_aws_object = issubclass(prop_type, BaseAWSObject)
+                # prop_type isn't a class
+                except TypeError:
+                    pass
+                if is_aws_object:
+                    v = kwargs[prop_name]
```

Comment:
not a huge deal, but you already define `value = kwargs[prop_name]` above, i would just reuse that here